### DISCUSSION TITLE
Version 9.6

### DISCUSSION THREAD

**Discussion by FilipePS** [FilipePS/Traduzir-paginas-web#486] [Announcements]
Added:
  - Option to replace selected text with translated text
    - When you type a message, select the message and translate the selected text, instead of the copy text button there will be a button to replace/paste the translated text with the selected text.
    - If the `Translate selected text when pressing Ctrl twice` option is enabled. Hitting `Ctrl` when the popup is open will be the same as clicking that new button.
    - When selecting text, the shortcut <kbd>Ctrl+Alt+X</kbd> will translate and replace the selected text instantly without showing any popup.
      - You can edit this hotkey in the extension's settings.</li>
  - Added 33 new target languages: <i>Assamese, Aymara, Bambara, Bashkir, Bhojpuri, Chuvash, Dhivehi, Dogri, Ewe, Guarani, Hill Mari, Ilocano, Kazakh (Latin), Konkani, Krio, Kurdish (Sorani), Lingala, Luganda, Mari, Maithili, Meiteilon (Manipuri), Mizo, Oromo, Papiamento, Quechua, Sanskrit, Sepedi, Tigrinya, Tsonga, Twi, Udmurt, Uzbek (Cyrillic), Yakut.</i>
  - Translation for language names in: <i>Catalan, Danish, Greek, Hungarian, Icelandic, Korean, Persian, Swedish, Thai and Portuguese (Portugal).</i></li>

Fixed:
  - Some parts of the translation are lost and consequently parts of the page are left untranslated
  - The <i><code>&lt;option&gt;</code></i> html tag is not translated.
  - Fixed: <i>showOriginalTextWhenHovering</i> always translates placeholder instead of value
  - Traditional Chinese is not detected as site language
  - text-to-speech does not work on long texts in some languages
  - Option to reset/import settings does not work in firefox android
  - The <b>original</b> button is blue even when the page is translated on android
  - Bugs related to invalid language code
  - Bugs related to translation with DeepL

Changes:
  - Now text-to-speech has an in-memory cache
  - Now the translation service will change silently if the selected target language is not supported by the preferred translation service.
  - Updated interface translations

<hr /><em>This discussion was created from the release <a href='https://github.com/FilipePS/Traduzir-paginas-web/releases/tag/v9.6'>Version 9.6</a>.</em>

**Comment by aoham** [comment]
TWP - Translate Web Pages latest version 9.6 does not work on firefox 78.12, previous version 9.5.4 works fine.

**Reply by verifyurhuman** [reply]
@aoham 
Version 9.6.1 add support for older versions of Firefox.

**Reply by aoham** [reply]
Thanks FilipePS, working now.

**Comment by 13112188160** [comment]
TWP - 翻译网页最新版本 9.6 在 谷歌浏览器上点击翻译没反应。

**Reply by verifyurhuman** [reply]
https://bookfere.com/post/1020.html

**Comment by GZ-Horse** [comment]
Can u join other translation engines, such as Microsoft Translate? Google Translate is now unavailable in Chinese mainland, thank you developer
